tools/runners/run-test-suite.py

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/env python
+
+# Copyright JS Foundation and other contributors, http://js.foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+import argparse
+import os
+import subprocess
+import sys
+
+import util
+
+def get_arguments():
+    execution_runtime = os.environ.get('RUNTIME')
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-q', '--quiet', action='store_true',
+                        help='Only print out failing tests')
+    parser.add_argument('--runtime', metavar='FILE', default=execution_runtime,
+                        help='Execution runtime (e.g. qemu)')
+    parser.add_argument('--engine', metavar='FILE',
+                        help='JerryScript binary to run tests with')
+    parser.add_argument('--test-list', metavar='FILE',
+                        help='File contains test paths to run')
+    parser.add_argument('--skip-list', metavar='LIST',
+                        help='Add a comma separated list of patterns of the excluded JS-tests')
+    parser.add_argument('--test-dir', metavar='DIR',
+                        help='Directory contains tests to run')
+    parser.add_argument('--snapshot', action='store_true',
+                        help='Snapshot test')
+
+    script_args = parser.parse_args()
+    if script_args.skip_list:
+        script_args.skip_list = script_args.skip_list.split(',')
+    else:
+        script_args.skip_list = []
+
+    return script_args
+
+
+def get_tests(test_dir, test_list, skip_list):
+    tests = []
+    if test_dir:
+        tests = []
+        for root, _, files in os.walk(test_dir):
+            tests.extend([os.path.join(root, test_file) for test_file in files if test_file.endswith('.js')])
+
+    if test_list:
+        with open(test_list, "r") as test_list_fd:
```

Comment:
Afaik this could be rewritten as:
```js
dirname = os.path.dirname(test_list)
with open(...) as test_list_fd:
  for entry in test_list_fd:
    tests.append(os.path.normpath(os.path.join(dirname, entry)))
```

Reply:
I checked, we can do it with an additional rstrip() for entry to strip newline character(s)